backend/src/services/trading_service.py

```
@@ -175,57 +195,141 @@ async def execute_sell_order(self, token_address: str, amount_tokens: float, sli
+            try:
+                response = await self.solana_client.get_signature_statuses([signature])
+                if response.value and response.value[0]:
+                    status = response.value[0]
+                    if status.confirmations is not None or status.err is None:
+                        return True
+            except Exception as e:
+                logger.debug(f"Error polling signature status: {e}")
+            await asyncio.sleep(interval)
+        return False
+
+    async def place_limit_order(self, token_address: str, target_price: float, amount_sol: float, side: str = 'buy', token_symbol: str = None) -> Dict:
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)